drivers/can/Kconfig.stm32

```diff
@@ -9,6 +9,7 @@
 config CAN_STM32
 	bool "STM32 CAN Driver"
 	depends on SOC_FAMILY_STM32
+	default y
```

Comment:
typically we do this per-board instead. Like for example: 
https://github.com/zephyrproject-rtos/zephyr/blob/master/boards/arm/stm32f4_disco/Kconfig.defconfig#L22

Reply:
This only selects the stm32 CAN driver if CAN is enabled in general. It's annoying to select CAN and the select the hardware driver. For example if you want to run the API tests it's helpful.

Reply:
Sure, but what about SoCs in the `SOC_FAMILY_STM32` that don't support CAN at all?

Reply:
This series won't set CONFIG_CAN to y

Reply:
I believe this is already done in `soc/arm/st_stm32/common/Kconfig.defconfig.series`, isn't it?

Reply:
@ydamigos you are right!